src/main/java/cubicchunks/regionlib/MultiUnsupportedDataException.java

```
@@ -21,17 +21,20 @@
  *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  *  THE SOFTWARE.
  */
-package cubicchunks.regionlib.util;
+package cubicchunks.regionlib;
 
-public class WrappedException extends RuntimeException {
+import java.io.IOException;
+import java.util.Map;
 
-	private final Throwable t;
+public class MultiUnsupportedDataException extends IOException {
+    private final Map<?, UnsupportedDataException> children;
 
-	public WrappedException(Throwable t) {
-		this.t = t;
-	}
+    public MultiUnsupportedDataException(Map<?, UnsupportedDataException> children) {
+        this.children = children;
+    }
```

Use `addSuppressed` instead of adding children yourself

---

but `SaveSectionException` does the same thing... i guess i should make it print them, though.

---

Yes, I didn't know about suppressed exceptions when I wrote it originally.

---

the thing is, i'm not suppressing these exceptions - i actually need them to be in a `Map`.

---

You could wrap each of them and specify the key it's for in a message. And from javadoc of addSuppressed:
```
Appends the specified exception to the exceptions that were suppressed in order to deliver this exception. This method is thread-safe and typically called (automatically and implicitly) by the try-with-resources statement.
```
You are suppressing them, to later throw a different exception.

---

i guess i could add an optional `key` parameter to `UnsupportedDataException`, but that seems even more hacky imo